v2/client/client.go

```diff
@@ -149,18 +149,17 @@ func (c *ceClient) Request(ctx context.Context, e event.Event) (*event.Event, pr
 			}
 		}()
 	}
+	if protocol.IsNACK(err) {
```

Comment:
I think what you should check here is `!protocol.IsACK(err)`. err could be either ack, nack or none of both, which means error even before sending

Reply:
yeah, NACK means only NACK, not rando error.

Reply:
@n3wscott OT but should we have a more comprehensive method like `protocol.IsFailed(res)`?

Reply:
Sure, we can add more helpers as we find them useful.